example-extension/.github/workflows/ci.yml

```diff
@@ -1,73 +1,110 @@
+            matrix:
+                php_version: {}
+    code-quality:
+        name: 'Code quality checks'
+        runs-on: ubuntu-22.04
+        steps:
+            -
+                name: Checkout
+                uses: actions/checkout@v4
+            -
+                name: 'Install PHP'
+                uses: shivammathur/setup-php@v2
+                with:
+                    php_version: '${{ matrix.php_version }}'
+                    coverage: none
+                    tools: 'composer:v2.4'
+            -
+                name: 'Show Composer version'
+                run: 'composer --version'
+            -
+                name: 'Show the Composer configuration'
+                run: 'composer config --global --list'
+            -
+                name: 'Cache dependencies installed with composer'
+                uses: actions/cache@v3
+                with:
+                    key: 'php${{ matrix.php_version }}-typo3${{ matrix.php_version }}'
+                    path: ~/.cache/composer
+                    restore-keys: "php${{ matrix.php_version }}-composer-\n"
+            -
+                name: 'Install Composer dependencies'
+                run: "composer require typo3/cms-core ^${{ matrix.typo3_version }} --no-progress\ncomposer install --no-progress\n"
+            -
+                name: 'Run command'
+                run: "composer ci:composer:psr-verify\ncomposer ci:json:lint\ncomposer ci:php:rector\ncomposer ci:php:cs-fixer\ncomposer ci:php:stan\n"
+        strategy:
+            fail-fast: false
+            matrix:
+                php_version: {}
+                typo3_version: {}
+    code-tests:
+        name: 'Coverage tests'
+        runs-on: ubuntu-22.04
+        steps:
+            -
+                name: Checkout
+                uses: actions/checkout@v4
+            -
+                name: 'Install PHP'
+                uses: shivammathur/setup-php@v2
+                with:
+                    php_version: '${{ matrix.php_version }}'
+                    coverage: none
+                    tools: 'composer:v2.4'
+            -
+                name: 'Show Composer version'
+                run: 'composer --version'
+            -
+                name: 'Show the Composer configuration'
+                run: 'composer config --global --list'
+            -
+                name: 'Cache dependencies installed with composer'
+                uses: actions/cache@v3
+                with:
+                    key: 'php${{ matrix.php_version }}-typo3${{ matrix.php_version }}'
+                    path: ~/.cache/composer
+                    restore-keys: "php${{ matrix.php_version }}-composer-\n"
+            -
+                name: 'Install Composer dependencies'
+                run: "composer require typo3/cms-core ^${{ matrix.typo3_version }} --no-progress\ncomposer install --no-progress\n"
+            -
+                name: 'Run command'
+                run: "composer ci:coverage\n"
```

Comment:
@mkarulin Is there a reason why we run `ci:coverage` instead of `ci:tests`? I am assuming that the coverage check is also running the tests and in addition does the coverage check. But do we do anything with the coverage check? Is there a threshold and can it fail?

Reply:
coverage is TBD